packages/amplify-graphql-api-construct-tests/src/sql-tests-common/sql-models-auto-increment.ts

```diff
@@ -0,0 +1,174 @@
+import * as path from 'path';
+import * as fs from 'fs-extra';
+import { ImportedRDSType } from '@aws-amplify/graphql-transformer-core';
+import AWSAppSyncClient, { AUTH_TYPE } from 'aws-appsync';
+import { createNewProjectDir, deleteProjectDir, getRDSTableNamePrefix } from 'amplify-category-api-e2e-core';
+import { initCDKProject, cdkDeploy, cdkDestroy } from '../commands';
+import { SqlDatatabaseController } from '../sql-datatabase-controller';
+import { CRUDLHelper } from '../utils/sql-crudl-helper';
+import { ONE_MINUTE } from '../utils/duration-constants';
+import { coffeeQueueFieldMap } from './schemas/sql-auto-increment/field-map';
+
+export const testGraphQLAPIAutoIncrement = (
+  options: {
+    projFolderName: string;
+    region: string;
+    connectionConfigName: string;
+    dbController: SqlDatatabaseController;
+    resourceNames: { sqlLambdaAliasName: string };
+  },
+  testBlockDescription: string,
+  engine: ImportedRDSType,
+): void => {
+  describe(`${testBlockDescription} - ${engine}`, () => {
+    // In particular, we want to verify that the new CREATE operation
+    // is allowed to omit the primary key field, and that the primary key
+    // we get back is the correct, db generated value.
+    // NOTE: Expects underlying orderNumber column to be a serial primary key in Postgres table
+    const schemaPath = path.resolve(path.join(__dirname, '..', 'sql-tests-common', 'schemas', 'sql-auto-increment', 'schema.graphql'));
+    const schemaConfigString = fs.readFileSync(schemaPath).toString();
+    const { projFolderName, region, connectionConfigName, dbController } = options;
+    const templatePath = path.resolve(path.join(__dirname, '..', '__tests__', 'backends', 'sql-models'));
+
+    let projRoot: string;
+    let name: string;
+    let outputs: Promise<any>;
+    let coffeeQueueTableCRUDLHelper: CRUDLHelper;
+
+    beforeAll(async () => {
+      projRoot = await createNewProjectDir(projFolderName);
+      name = await initCDKProject(projRoot, templatePath);
+      dbController.writeDbDetails(projRoot, connectionConfigName, schemaConfigString);
+      outputs = await cdkDeploy(projRoot, '--all', { postDeployWaitMs: ONE_MINUTE });
+      const { awsAppsyncApiEndpoint: apiEndpoint, awsAppsyncApiKey: apiKey } = outputs[name];
+
+      const appSyncClient = new AWSAppSyncClient({
+        url: apiEndpoint,
+        region,
+        disableOffline: true,
+        auth: {
+          type: AUTH_TYPE.API_KEY,
+          apiKey,
+        },
+      });
+
+      coffeeQueueTableCRUDLHelper = new CRUDLHelper(appSyncClient, 'CoffeeQueue', 'CoffeeQueues', coffeeQueueFieldMap);
```

Comment:
The only change since previous approval is the use of `coffeeQueueFieldMap` which is necessary after changes merged in #2906.